github-actions/scan/__test__/sechub-cli.test.ts

```diff
@@ -270,4 +300,104 @@ describe('defineFalsePositives', function () {
         expect(context.lastClientExitCode).toBe(0);
         expect(context.defineFalsePositivesFile).toBe(file);
     });
+});
+
+describe('spawnAndWait signal handling', () => {    
+    let mockChildProcess: any;
+
+    beforeEach(() => {
+        // Create a mock child process object
+        mockChildProcess = {
+            kill: jest.fn(),
+            on: jest.fn(),
+        };
+
+        // Set the mock return value for spawn
+        (spawn as jest.Mock).mockReturnValue(mockChildProcess);
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    it('should forward SIGINT signal to child process', async () => {
+        /* prepare */
+        const command = 'dummyCommand';
+        const args = ['dummyArg'];
+        const options = {};
+
+        /* execute */
+        const promise = spawnAndWait(command, args, options);
+
+        // Emit SIGINT signal
+        expect(process.emit('SIGINT', 'SIGINT'));
+
+        /* test */
+        // Verify that the signal was forwarded
+        expect(mockChildProcess.kill).toHaveBeenCalledWith('SIGINT');
+
+        // Simulate child process exit
+        mockChildProcess.on.mock.calls[0][1](0, null);
+
+        await expect(promise).resolves.toBe(0);
+    });
+
+    it('should forward SIGTERM signal to child process', async () => {
+        /* prepare */
+        const command = 'dummyCommand';
+        const args = ['dummyArg'];
+        const options = {};
+
+        /* execute */
+        const promise = spawnAndWait(command, args, options);
+
+        // Emit SIGTERM signal
+        expect(process.emit('SIGTERM', 'SIGTERM'));
+
+        /* test */
+        // Verify that the signal was forwarded
+        expect(mockChildProcess.kill).toHaveBeenCalledWith('SIGTERM');
+
+        // Simulate child process exit
+        mockChildProcess.on.mock.calls[0][1](0, null);
+
+        await expect(promise).resolves.toBe(0);
+    });
+
+    it('should reject promise if child process is terminated by signal', async () => {
+        /* prepare */
+        const command = 'dummyCommand';
+        const args = ['dummyArg'];
+        const options = {};
+
+        /* execute */
+        const promise = spawnAndWait(command, args, options);
+
+        // Emit SIGTERM signal
+        expect(process.emit('SIGTERM', 'SIGTERM'));
+
+        /* test */
+        // Verify that the signal was forwarded
+        expect(mockChildProcess.kill).toHaveBeenCalledWith('SIGTERM');
+
+        // Simulate child process exit with signal
+        mockChildProcess.on.mock.calls[0][1](null, 'SIGTERM');
+
+        await expect(promise).rejects.toThrow('Process terminated by signal: SIGTERM');
+    });
+
+    it('should reject promise if child process encounters an error', async () => {
```

Comment:
You have written tests for
- reject on error
- reject on sigterm
- forward SIGTERM
- forward SIGINT

But what is not tested is, that
- no reject when normal exit

but maybe this is not necessary, because we have the integration tests.